### PR TITLE
Cinema Support + MODS Split

### PR DESCRIPTION
**Pull request by Nasafrasa**
Adds support for adding and manipulating a Cinema Screen! (with docs!!!)

Example Syntax:

```ts
const screen = new CinemaScreen("YoutubeID", "dQw4w9WgXcQ");
screen.colorCorrection.brightness = 1.5;
screen.colorCorrection.saturation = 1.25;
screen.author = "nasafrasa";
screen.videoFile = "ImportantVideo";
screen.height = 100;
screen.push();
```

### CODE REVIEW

File: src/cinema.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    set mergePropGroups(value: boolean) { this.json.mergePropGroups = value }
+    set transparency(value: boolean) { this.json.transparency = value }
+    set bloom(value: number) { this.json.bloom = value }
+
+    set colorCorrection(value: ColorCorrection) { this.json.colorCorrection = value }
+    set vignette(value: Vignette) { this.json.vignette = value }
+
+}
+
+type ColorCorrection = {
```
Comment (Fernthedev) [inline src/cinema.ts:151]: Usually I use interface for this, but that's just me being a nitpick

File: src/constants.ts [hunk 2]
```diff
@@ -21,7 +21,8 @@ export type DIFFS =
 /** All mods to require/suggest. */
 export type MODS =
     "Chroma" |
-    "Noodle Extensions"
+    "Noodle Extensions" |
+    "Cinema"
```
Comment (Fernthedev) [inline src/constants.ts:25]: Something should enforce this so it's only allowed in `suggestions`

File: src/cinema.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    /**
+     * Push Main Cinema Screen to cinema-video.json
+     * @returns 
+     */
+    push() {
+        // Warnings for out of range numbers for certain Cinema Properties
+        function cinemaRange(thing: string, number: number) {
+            console.log("\x1b[31m", `Cinema Warning: ${thing} value`, "\x1b[31m", number, "\x1b[31m", "out of range!",  "\x1b[0m")
```
Comment (Fernthedev) [inline src/cinema.ts:48]: Any chance you could use constants/variables for the ASCII color coding for clarity?

File: src/cinema.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
+// deno-lint-ignore-file no-explicit-any adjacent-overload-signatures
+import { Json } from './beatmap.ts';
+import { jsonGet } from './general.ts';
+
+export class CinemaScreen {
+    json: Record<string, any> = {
```
Comment (Fernthedev) [inline src/cinema.ts:6]: It would be preferable if the json type was strongly typed instead of `Record<string, any>`

Comment (Nasafrasa) [reply src/cinema.ts]: 👍

File: src/cinema.ts (hunk 3 above)
Comment (Nasafrasa) [reply src/cinema.ts]: 👍

File: src/constants.ts (hunk 2 above)
Comment (Nasafrasa) [reply src/constants.ts]: 👍

**Pull request merged**